File: python/Queue.py
from threading import Condition

class Queue() :

    def __init__(self, capacite, l_liste, l_taille) :
        self.valeurs = []
        self.capacite = capacite
        self.l_liste = l_liste
        self.l_taille = l_taille
        self.verrou = Condition()
        self.l_liste.config(text="File = "+self.createString())


    def majLabel(self) :
        self.l_liste.config(text="File = "+self.createString())
        self.l_liste.update()
        self.l_taille.update()

    def defiler(self) :
        with self.verrou :
            while len(self.valeurs) == 0 : self.verrou.wait()
            n = self.valeurs[0]
            del self.valeurs[0]
            self.majLabel()
            self.verrou.notifyAll()
            return n

    def enfiler(self, n) :
        with self.verrou :
            while len(self.valeurs) == self.capacite : self.verrou.wait()
            self.valeurs.append(n)
            self.majLabel()
            self.verrou.notifyAll()

    def createString(self) :
        with self.verrou :
            s = "<<< "
            for i in range(0,len(self.valeurs)) :
              s+=str(self.valeurs[i])
              if i != len(self.valeurs)-1 :  s+=", "
            s+=" <<<"
            self.l_taille.config(text="Longueur de la file : "+ str(len(self.valeurs)) );
            return s;

File: java/Queue.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import java.util.LinkedList;

public class Queue{
  private final int capacite;
  private final LinkedList<Integer> valeurs = new LinkedList<Integer>();
  private final JLabel l_liste, l_taille;

  public Queue(int capacite, JLabel l_liste, JLabel l_taille){
    this.capacite = capacite;
    this.l_liste = l_liste;
    this.l_taille = l_taille;
    this.l_liste.setText("File = "+this.createString());
  }
  //[Mise à jouer du text des JLabels
  private void majLabel(){
    this.l_liste.setText("File = "+this.createString());
    this.l_liste.repaint();
    this.l_taille.repaint();
  }
  //]

  //[Suppression du premier élément de la liste
  public synchronized int defiler() throws InterruptedException {
    while(this.valeurs.size() == 0) wait();
    int n = this.valeurs.removeFirst();
    this.majLabel();
    notifyAll();
    return n;
  }
  //]

  //[Ajout d'un élément à la liste
  public synchronized void enfiler(int n) throws InterruptedException{
    while(this.valeurs.size() == this.capacite) wait();
    this.valeurs.add(n);
    this.majLabel();
    notifyAll();
  }
  //]

  //[Création de la chaîne de caractères de la liste + longueur de la liste
  public synchronized String createString(){
    String s = "<<< ";
    for(int i=0; i<this.valeurs.size(); i++){
      s+=String.valueOf(this.valeurs.get(i));
      if(i!=this.valeurs.size()-1) s+=", ";
    }
    s+=" <<<";
    this.l_taille.setText("Longueur de la file : "+this.valeurs.size());
    return s;
  }
  //]
}
